File: Shashank Shukla 18-CSE/Day1.java
// My code will be here T = O(N), S = O(N)
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Integer> ss = new Stack<>();
        Stack<Integer> tt = new Stack<>();
        
        for(int i = 0; i<s.length(); i++){
            if(s.charAt(i) != '#')
                ss.push((int)s.charAt(i));
            else{
                if(!ss.isEmpty())
                    ss.pop();
            }
        }
        for(int i = 0; i<t.length(); i++){
            if(t.charAt(i) != '#')
                tt.push((int)t.charAt(i));
            else {
                if(!tt.isEmpty())
                    tt.pop();
            }
        }
        if(ss.size() != tt.size())
            return false;
        while(!ss.isEmpty()){
            if(ss.peek() != tt.peek())
                return false;
            ss.pop();
            tt.pop();
        }
        return true;
    }
}

File: Yatindra_19GCEBC004/day_1.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string s1="";
        string t1="";
        stack <char> s_char;
        stack <char> t_char;
        for(int i=0;i<s.length();i++){
            if(s[i]=='#' && !s_char.empty()){
                s_char.pop();
            }
            else if(s[i]!='#'){
                s_char.push(s[i]);
            }
        }
        for(int j=0;j<t.length();j++){
            if(t[j]=='#' && !t_char.empty()){
                t_char.pop();
            }
            else if(t[j]!='#'){
                t_char.push(t[j]);
            }
        }
        while(!s_char.empty()){
            s1=s_char.top()+s1;
            s_char.pop();
        }
        while(!t_char.empty()){
            t1=t_char.top()+t1;
            t_char.pop();
        }
        if(s1.compare(t1)==0){
            return true;
        }
        else{
            return false;
        }
    }
};